frontend/src/concepts/pipelines/apiHooks/mlmd/__tests__/useGetExecutionById.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { testHook, standardUseFetchState } from '~/__tests__/unit/testUtils/hooks';
+import { MetadataStoreServicePromiseClient } from '~/third_party/mlmd';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { useGetExecutionById } from '~/concepts/pipelines/apiHooks/mlmd/useGetExecutionById';
+
+// Mock the usePipelinesAPI hook
+jest.mock('~/concepts/pipelines/context', () => ({
+  usePipelinesAPI: jest.fn(),
+}));
+
+// Mock the MetadataStoreServicePromiseClient
+jest.mock('~/third_party/mlmd', () => {
+  const originalModule = jest.requireActual('~/third_party/mlmd');
+  return {
+    ...originalModule,
+    MetadataStoreServicePromiseClient: jest.fn().mockImplementation(() => ({
+      getExecutionsByID: jest.fn(),
+    })),
+    GetExecutionsByIDRequest: originalModule.GetExecutionsByIDRequest,
+  };
+});
+
+describe('useGetExecutionById', () => {
+  const mockClient = new MetadataStoreServicePromiseClient('');
+  const mockUsePipelinesAPI = usePipelinesAPI as jest.Mock;
+  const mockGetExecutionsByID = mockClient.getExecutionsByID as jest.Mock;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockUsePipelinesAPI.mockReturnValue({
+      metadataStoreServiceClient: mockClient,
+    });
+  });
+
+  it('should return null if no executionId is provided', async () => {
+    const renderResult = testHook(useGetExecutionById)();
+
+    expect(renderResult.result.current).toStrictEqual(standardUseFetchState(null));
+    expect(renderResult).hookToHaveUpdateCount(1);
+  });
+
+  it('should return null if executionId is NaN', async () => {
+    const renderResult = testHook(useGetExecutionById)('not-a-number');
+
+    expect(renderResult.result.current).toStrictEqual(standardUseFetchState(null));
+    expect(renderResult).hookToHaveUpdateCount(1);
+  });
```

Comment:
You've only asserted the default value is returned by `useFetchState` because testing with `useFetchState` requires a call to `renderResult.waitForNextUpdate()` before the computed result is available.